exopy_hqc_legacy/tasks/tasks/instr/apply_mag_field_task.py

```diff
@@ -59,41 +59,47 @@ def perform(self, target_value=None):
 
         driver = self.driver
         normal_end = True
-        if (abs(driver.read_persistent_field() - target_value) >
-                driver.output_fluctuations):
+        # Only if driver.heater_state == 'Off' otherwise we wait for 
+        # post_switch_wait no matter what
+        if driver.heater_state == 'Off':
             job = driver.sweep_to_persistent_field()
             if job.wait_for_completion(self.check_for_interruption,
-                                       timeout=60, refresh_time=1):
+                                       timeout=60, refresh_time=3):
                 driver.heater_state = 'On'
                 sleep(self.post_switch_wait)
             else:
                 return False
 
+        if (abs(driver.read_persistent_field() - target_value) > 
+            driver.output_fluctuations):
             # set the magnetic field
             job = driver.sweep_to_field(target_value, self.rate)
             normal_end = job.wait_for_completion(self.check_for_interruption,
                                                  timeout=60,
                                                  refresh_time=10)
+            print('field:', driver.read_persistent_field())
```

Comment:
If we want to keep this it should be a logger call.

Reply:
No this was for debugging purpose I'll remove it